src/feature/control/control_getinfo.c

```diff
@@ -326,46 +326,37 @@ getinfo_helper_current_time(control_connection_t *control_conn,
 }
 
 /**
- * Switch between microdesc vs networkstatus descriptor dumps
- * Assumes that question would be either:
- * - dir/status-vote/current/consensus-microdesc
- * or
- * - dir/status-vote/current/consensus
+ * Helps to access directory information for a given
+ * type of consensus.
+ *
+ * returns: int
+ *   -1 in cases of an error
+ *   0 otherwise
  */
 STATIC int
-getinfo_helper_current_consensus(const char* question,
-                          char** answer,
-                          const char** errmsg)
+getinfo_helper_current_consensus(const char* consensus_type,
+                                 int flavor, char** answer,
+                                 const char** errmsg)
 {
-  // Ensures question contains a request for
-  // ns or microdesc consensus
-  if (
-      strcmp(question, "dir/status-vote/current/consensus") &&
-      strcmp(question, "dir/status-vote/current/consensus-microdesc")) {
-    return 0;
-  }
-
-  const char* consensus_type = !strcmp(
-     question, "dir/status-vote/current/consensus-microdesc"
-  ) ? "microdesc" : "ns";
-
-  if (we_want_to_fetch_flavor(get_options(), FLAV_NS)) {
+  /** Check if the consensus flavor is cached or in the ram. */
+  if (we_want_to_fetch_flavor(get_options(), flavor)) {
     const cached_dir_t *consensus = dirserv_get_consensus(consensus_type);
     if (consensus) {
       *answer = tor_strdup(consensus->dir);
     }
   }
-  if (!*answer) { /* try loading it from disk */
+  if (!*answer) { /* else, try loading consensus from disk */
     tor_mmap_t *mapped = networkstatus_map_cached_consensus(consensus_type);
     if (mapped) {
       *answer = tor_memdup_nulterm(mapped->data, mapped->size);
       tor_munmap_file(mapped);
+      tor_free(mapped);
```

Comment:
Why are you freeing this here? Are you sure this is right? Please double check this because I think you might be freeing the actual global consensus.

Reply:
The CI was complaining about memory leaks and that's why I tried to do this. I found that the problem was in the test itself where I wasn't freeing `answer` I'll undo this and check.